config/local_defaults.go

```diff
@@ -18,487 +18,178 @@ package config
+			case reflect.Bool:
+				if reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).Bool() == reflect.ValueOf(&defaultCurrentConfig).Elem().FieldByName(field.Name).Bool() {
+					// we're skipping the error checking here since we already tested that in the unit test.
+					boolVal, _ := strconv.ParseBool(nextVersionDefaultValue)
+					reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).SetBool(boolVal)
+				}
+			case reflect.Int32:
+				fallthrough
+			case reflect.Int:
+				fallthrough
+			case reflect.Int64:
+				if reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).Int() == reflect.ValueOf(&defaultCurrentConfig).Elem().FieldByName(field.Name).Int() {
+					// we're skipping the error checking here since we already tested that in the unit test.
+					intVal, _ := strconv.ParseInt(nextVersionDefaultValue, 10, 64)
+					reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).SetInt(intVal)
+				}
+			case reflect.Uint32:
+				fallthrough
+			case reflect.Uint:
+				fallthrough
+			case reflect.Uint64:
+				if reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).Uint() == reflect.ValueOf(&defaultCurrentConfig).Elem().FieldByName(field.Name).Uint() {
+					// we're skipping the error checking here since we already tested that in the unit test.
+					uintVal, _ := strconv.ParseUint(nextVersionDefaultValue, 10, 64)
+					reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).SetUint(uintVal)
+				}
+			case reflect.String:
+				if reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).String() == reflect.ValueOf(&defaultCurrentConfig).Elem().FieldByName(field.Name).String() {
+					// we're skipping the error checking here since we already tested that in the unit test.
+					reflect.ValueOf(&newCfg).Elem().FieldByName(field.Name).SetString(nextVersionDefaultValue)
+				}
+			default:
+				panic(fmt.Sprintf("unsupported data type (%s) encountered when reflecting on config.Local datatype %s", reflect.ValueOf(&defaultCurrentConfig).Elem().FieldByName(field.Name).Kind(), field.Name))
+			}
 		}
-		if newCfg.IncomingConnectionsLimit == defaultLocalV0.IncomingConnectionsLimit {
-			newCfg.IncomingConnectionsLimit = defaultLocalV1.IncomingConnectionsLimit
-		}
-		if newCfg.ReconnectTime == defaultLocalV0.ReconnectTime {
-			newCfg.ReconnectTime = defaultLocalV1.ReconnectTime
-		}
-		newCfg.Version = 1
-	}
-	// Migrate 1 -> 2
-	if newCfg.Version == 1 {
-		if newCfg.ReservedFDs == defaultLocalV1.ReservedFDs {
-			newCfg.ReservedFDs = defaultLocalV2.ReservedFDs
-		}
-		newCfg.Version = 2
 	}
-	// Migrate 2 -> 3
-	if newCfg.Version == 2 {
-		if newCfg.MaxConnectionsPerIP == defaultLocalV2.MaxConnectionsPerIP {
-			newCfg.MaxConnectionsPerIP = defaultLocalV3.MaxConnectionsPerIP
-		}
-		if newCfg.CatchupParallelBlocks == defaultLocalV2.CatchupParallelBlocks {
-			newCfg.CatchupParallelBlocks = defaultLocalV3.CatchupParallelBlocks
-		}
-		newCfg.Version = 3
+	return
+}
+
+func getLatestConfigVersion() uint32 {
```

Comment:
Does it also make sense to instead have an expected "latest config version" global variable, and have a unit test catch whether or not this is inconsistent with the tags?

Reply:
I'm trying to ensure that when we make changes to the `Local`, it need to be reflected in less places, which would make it more simple to use and maintain. Adding a global variable that need to be updated with any `Local` change would defeat that, wouldn't it ?

Reply:
Yeah, that's true. I did wonder whether this would be easier than adding another tag to `Local.Version`, i.e. `version[i]:"i"`, which is somewhat redundant as well; it seems like over time that the list of tags would get really long, and it doesn't seem to provide any extra information.